internal/pkg/composable/providers/kubernetessecrets/kubernetes_secrets_test.go

```diff
@@ -935,6 +937,69 @@ func Test_Config(t *testing.T) {
 	}
 }
 
+func Test_FetchFromAPI(t *testing.T) {
+	for _, tc := range []struct {
+		name                    string
+		k8sClient               k8sclient.Interface
+		secretName              string
+		secretNamespace         string
+		secretKey               string
+		expectedValue           string
+		expectedResourceVersion string
+		expectedOK              bool
+	}{
+		{
+			name:      "k8s client is nil",
+			k8sClient: nil,
+		},
+		{
+			name:            "secret not found",
+			k8sClient:       k8sfake.NewClientset(),
+			secretName:      "secret_name",
+			secretNamespace: "secret_namespace",
+			secretKey:       "secret_key",
+		},
+		{
+			name: "key in secret not found",
+			k8sClient: k8sfake.NewClientset(
+				buildK8SSecret("secret_namespace", "secret_name", "secret_key", "secret_value"),
+			),
+			secretName:      "secret_name",
+			secretNamespace: "secret_namespace",
+			secretKey:       "secret_key_not_found",
+		},
+		{
+			name: "key in secret not found",
```

Comment:
Isn't this the happy path case, where the key _is_ found?

Reply:
this is the almost happy path 😆  so the secret is found in the api server, there is no error reading it, but the key inside the secret we are trying to fetch doesn't exist. does it make sense?

Reply:
I must be missing something 😕 IIUC, the key we're looking for is `secret_key` (that's what we're passing to the `fetchFromAPI` function on line 978), and the mock secret we're building also has a key of `secret_key` (line 974). So to me it seems like the key is found

Reply:
https://github.com/elastic/elastic-agent/pull/7057 addressed in this PR 🙂